### PR TITLE
Implemented crdcdh-1136

### PR DESCRIPTION
**Pull request by vshand11**
Implemented crdcdh-1136

### CODE REVIEW

File: src/common/model_reader.py [hunk 1]
```diff
@@ -56,15 +56,15 @@
 valid_relationship_types = ["many_to_one", "one_to_one", "many_to_many"]
 
 
-def get_list_values(list_str):
-    return [item.strip() for item in list_str.split(LIST_DELIMITER) if item.strip()]
+def get_list_values(list_str, list_delimiter):
```
Comment (n2iw) [inline src/common/model_reader.py:59]: It seems this function is not used anywhere. If so, we can remove it.

File: src/common/model_reader.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ def __init__(self, yaml_files, data_common, version=DEFAULT_VERSION):
                 raise
 
         self.model.update({MODEL_SOURCE: model_file_src})
+        self.model.update({LIST_DELIMITER_PROP: delimiter if delimiter else ','})
```
Comment (n2iw) [inline src/common/model_reader.py:89]: The default delimiter should be pipe character "|"

File: src/metadata_validator.py [hunk 3]
```diff
@@ -482,7 +482,8 @@ def validate_prop_value(self, prop_name, value, prop_def, msg_prefix):
             
             elif type == "array" or type == "value-list":
                 val = str(value)
-                arr = val.split("*") if "*" in val else val.split(",") if "," in val else [value]
+                list_delimiter = self.model.get_list_delimiter()
+                arr = val.split(list_delimiter) if list_delimiter in val else val.split("*") if "*" in val else [value]
```
Comment (n2iw) [inline src/metadata_validator.py:486]: Since we have default list_delimiter already stored the model, there is no need to try to use "*" to split anymore.

**Review by n2iw — CHANGES_REQUESTED** [review]

File: src/common/model_reader.py (hunk 1 above)
Comment (vshand11) [reply src/common/model_reader.py:59]: Removed

File: src/common/model_reader.py (hunk 2 above)
Comment (vshand11) [reply src/common/model_reader.py:89]: Updated

File: src/metadata_validator.py (hunk 3 above)
Comment (vshand11) [reply src/metadata_validator.py:486]: Yes for list and value list.  But not sure if array type will also use the same delimiter.

Comment (vshand11) [reply src/metadata_validator.py:486]: Leave '*' delimiter now for array type now. I can update after get confirmed.

File: src/metadata_validator.py [hunk 4]
```diff
@@ -482,7 +482,8 @@ def validate_prop_value(self, prop_name, value, prop_def, msg_prefix):
             
             elif type == "array" or type == "value-list":
                 val = str(value)
-                arr = val.split("*") if "*" in val else val.split(",") if "," in val else [value]
+                list_delimiter = self.model.get_list_delimiter()
+                arr = val.split(list_delimiter) if list_delimiter in val else [value]
```
Comment (n2iw) [inline src/metadata_validator.py:486]: Do we need to strip the leading and trailing spaces here? In case we got "TXT | PDF | JPEG"

Comment (vshand11) [reply src/metadata_validator.py:486]: Yes, each item is stripped as shown below:
for item in arr:
                    val = item.strip() if item and isinstance(item, str) else item
                    result, error = check_permissive(val, permissive_vals, msg_prefix, prop_name)
                    if not result:
                        errors.append(error)

**Review by n2iw — APPROVED** [review]

**Pull request merged**